sdk/containerregistry/azure-containers-containerregistry/README.md

```diff
@@ -102,19 +125,169 @@ ContainerRegistryAsyncClient client = new ContainerRegistryClientBuilder()
 client.listRepositoryNames().subscribe(repository -> System.out.println(repository));
 ```
 
+### Delete Images
+
+<!-- embedme ./src/samples/java/com/azure/containers/containerregistry/ReadmeSamples.java#L93-L121 -->
+```Java
+TokenCredential defaultCredential = new DefaultAzureCredentialBuilder().build();
+
+ContainerRegistryClient client = new ContainerRegistryClientBuilder()
+    .endpoint(endpoint)
+    .credential(defaultCredential)
+    .buildClient();
+
+final int imagesCountToKeep = 3;
+for (String repositoryName : client.listRepositoryNames()) {
+    final ContainerRepository repository = client.getRepository(repositoryName);
+
+    // Obtain the images ordered from newest to oldest
+    PagedIterable<ArtifactManifestProperties> imageManifests =
+        repository.listManifests(
+            ManifestOrderBy.LAST_UPDATED_ON_DESCENDING,
+            Context.NONE);
+
+    imageManifests.stream().skip(imagesCountToKeep)
+        .forEach(imageManifest -> {
+            System.out.printf(String.format("Deleting image with digest %s.%n", imageManifest.getDigest()));
+            System.out.printf("    This image has the following tags: ");
+
+            for (String tagName : imageManifest.getTags()) {
+                System.out.printf("        %s:%s", imageManifest.getRepositoryName(), tagName);
+            }
+
+            repository.getArtifact(imageManifest.getDigest()).delete();
+        });
+}
+```
+
+### Delete images asynchronously
+
+<!-- embedme ./src/samples/java/com/azure/containers/containerregistry/ReadmeSamples.java#L125-L151 -->
+```Java
+TokenCredential defaultCredential = new DefaultAzureCredentialBuilder().build();
+
+ContainerRegistryAsyncClient client = new ContainerRegistryClientBuilder()
+    .endpoint(endpoint)
+    .credential(defaultCredential)
+    .buildAsyncClient();
+
+final int imagesCountToKeep = 3;
+client.listRepositoryNames().subscribe(repositoryName -> {
+    final ContainerRepositoryAsync repository = client.getRepository(repositoryName);
+
+    PagedFlux<ArtifactManifestProperties> imageManifests = repository.listManifests(
+        ManifestOrderBy.LAST_UPDATED_ON_DESCENDING);
+
+    imageManifests.skip(imagesCountToKeep).subscribe(imageManifest -> {
+        System.out.printf(String.format("Deleting image with digest %s.%n", imageManifest.getDigest()));
+        System.out.printf("    This image has the following tags: ");
+
+        for (String tagName : imageManifest.getTags()) {
+            System.out.printf("        %s:%s", imageManifest.getRepositoryName(), tagName);
+        }
+
+        repository.getArtifact(imageManifest.getDigest()).delete().subscribe();
```

Comment:
Generally, having a `subscribe()` within another `subscribe()` is not a recommended pattern. Instead use `flatMap()`.

```java
client.listRepositoryNames()
         .map(repositoryName -> client.getRepository(repositoryName))
         .flatMap(repository -> repository.listManifests(..).skip(3))
         .flatMap(manifest -> { // process manifest})
         .subscribe();   
```